### PR TITLE
Reports Boilerplate

### LINKED ISSUE #506: Create the reports feature
## What needs to be done?
- [x] Create the reports feature scaffolding and inject the reports service and time service into it
- [x] Add a start and end date properties to the reports state and set their default values to contain the last seven days
- [x] Add a currently selected workspace property to the reports state and set it to user's current WS id
- [x] Add a report totals and reports summary properties to the reports state. They should be `Loadable`s.

### Dependencies
Depends on #503, #504 and #505

### LINKED ISSUE #507: Create the ReportsShortcut enum
## What needs to be done?
- [x] Create a `ReportsShortcut` enum with the following cases: 
```
    today,
    yesterday,
    thisWeek,
    lastWeek,
    thisMonth,
    lastMonth,
    thisYear,
    lastYear
```
- [x] Add a `localizedDescription` property that returns a localised name for the shortcut
- [x] Add a `dateRange(for now: Date)` function that returns a start/end date tuple for a current date (e.g. `.today` should return the same date, `.yesterday` — the day before, etc)

### Dependencies
Depends on #506

### LINKED ISSUE #508: Create the reports SelectionSource enum
## What needs to be done?
- [ ] Create a `SelectionSource` enum with the following cases:
```
    initial,
    calendar,
    shortcut(ReportsShortcut),
    siri
```

### Dependencies
Depends on #507

### LINKED ISSUE #511: Create the DateRangeSelection struct for reports
## What needs to be done?
- [ ] Create a `DateRangeSelection` struct, it should contain a start date, an end date, and a `SelectionSource`

### Dependencies
Depends on #508

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
This adds the Reports boilerplate

### Technical considerations
Nothing to see here; This merely adds things so we can work on it in more detail later

### Relationships
Refs #504
Closes #506
Closes #507
Closes #508
Closes #511

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Go at it commit by commit

### Review template
- [X] The code is thoroughly tested in well written and resilient manner
- [X] The code is performant and leads to an optimal user experience
- [X] The code is testable and does not cut corners for the sake of delivering
- [X] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
Anyone, anytime

### CODE REVIEW

File: Modules/Reports/Reports/Reports.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  Reports.h
+//  Reports
+//
+//  Created by William Barbosa on 03/08/20.
+//  Copyright © 2020 Toggl. All rights reserved.
+//
```
Comment (Odrakir) [inline Modules/Reports/Reports/Reports.h:7]: 🔥

File: Modules/Reports/Project.debug.xcconfig [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "../../App/Configurations/base/Project-Base.debug.xcconfig"
```
Comment (Odrakir) [inline Modules/Reports/Project.debug.xcconfig:1]: When adding a new framework, on top of setting the xcconfig files which will set the build settings, you have to delete the customized settings for both targets. So select the framework, select one target, build settings, filter to show only customize fields, select all (cmd + A) and delete. Then do the same with the other target.

That way all build setting will come from the xcconfig files.

File: Modules/Reports/ReportsTests/ReportsShortcutTests.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        XCTAssertEqual(range.end, ReportsShortcutTests.dateFrom(year: 2018, month: 12, day: 31))
+    }
+    
+    private static func dateFrom(year: Int, month: Int, day: Int) -> Date {
+        let calendar = Calendar(identifier: .gregorian)
+        let components = DateComponents(year: year, month: month, day: day, hour: 0, minute: 0, second: 0)
+        return calendar.date(from: components)!.ignoreTimeComponents()
+    }
+
+//    func dateRange(from now: Date, _ beginningOfWeek: DayOfTheWeek) -> (start: Date, end: Date?) {
```
Comment (sofokli1) [inline Modules/Reports/ReportsTests/ReportsShortcutTests.swift:72]: 🔥

File: Modules/Reports/Reports.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -560,94 +560,29 @@
...
 			buildSettings = {
-				CLANG_ENABLE_MODULES = YES;
-				CODE_SIGN_STYLE = Automatic;
-				DEFINES_MODULE = YES;
-				DYLIB_COMPATIBILITY_VERSION = 1;
-				DYLIB_CURRENT_VERSION = 1;
-				DYLIB_INSTALL_NAME_BASE = "@rpath";
-				INFOPLIST_FILE = Reports/Info.plist;
-				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (Odrakir) [inline Modules/Reports/Reports.xcodeproj/project.pbxproj:563]: This one needs :fire: too

Comment (heytherewill) [reply Modules/Reports/Reports.xcodeproj/project.pbxproj:563]: The app didn't build for me locally when I removed it 🤔 
Do I need to do something else?

Comment (sofokli1) [reply Modules/Reports/Reports.xcodeproj/project.pbxproj]: sometimes quitting xcode can help. next step clean project, delete derived data, and restart xcode. if all of these do not work, maybe we need those changes

**Pull request merged**